tests/AutoReview/ProjectCodeTest.php

```diff
@@ -0,0 +1,404 @@
+     * This list should never be added to, only removed from.
+     *
+     * @var string[]
+     */
+    private static $nonFinalNonExtensionClasses = [
+        ConsoleHelper::class,
+        SourceDirGuesser::class,
+        InfectionConfig::class,
+        DiffColorizer::class,
+        Differ::class,
+        EventDispatcher::class,
+        TestFrameworkFinder::class,
+        BadgeApiClient::class,
+        MetricsCalculator::class,
+        ProcessBuilder::class,
+        CodeCoverageData::class,
+        PhpSpecInitalConfigBuilder::class,
+        PhpUnitInitalConfigBuilder::class,
+        PhpSpecMutationConfigBuilder::class,
+        PhpUnitMutationConfigBuilder::class,
+        XmlConfigurationHelper::class,
+        CoverageXmlParser::class,
+        VersionParser::class,
+    ];
+
+    public function test_infection_bin_is_executable()
+    {
+        $infectionFile = __DIR__ . '/../../bin/infection';
+        $this->assertFileExists($infectionFile);
+        $this->assertTrue(is_executable($infectionFile));
+    }
+
+    /**
+     * @dataProvider providesSourceClasses
+     *
+     * @param string $className
+     */
+    public function test_src_class_provider_is_valid(string $className)
+    {
+        $this->assertTrue(
+            class_exists($className) || interface_exists($className) || trait_exists($className),
+            sprintf(
+                'The "%s" class was picked up by the source files finder, but it is not a class, interface or trait. ' .
+                'Please check for typos in the class name. Or exclude the file if in the ProjectCodeTest if it is not a class.',
+                $className
+            )
+        );
+    }
+
+    /**
+     * @dataProvider providesTestClassCases
+     *
+     * @param string $className
+     */
+    public function test_test_class_provider_is_valid(string $className)
+    {
+        $this->assertTrue(
+            class_exists($className) || interface_exists($className) || trait_exists($className),
+            sprintf(
+                'The "%s" class was picked up by the test files finder, but it is not a class, interface or trait. ' .
+                'Please check for typos in the class name. Or exclude the file if in the ProjectCodeTest if it is not a class.',
+                $className
+            )
+        );
+    }
+
+    /**
+     * @dataProvider providesSourceClasses
+     *
+     * @param string $className
+     */
+    public function test_non_extension_points_are_internal(string $className)
+    {
+        $rc = new \ReflectionClass($className);
+        $docBlock = $rc->getDocComment();
+
+        if (in_array($className, self::$extensionPoints)) {
+            if ($docBlock === false) {
+                $this->markTestSkipped(
+                    sprintf(
+                        'The "%s" class is an extension point, but does not have a doc-block.' .
+                        'Consider adding a doc-block to improve usability.',
```

Comment:
please add a space before `Consider`, now it looks like `...a doc-block.Consider...`

Reply:
:+1:  fixed it.